presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PruneUnreferencedOutputs.java

```diff
@@ -759,5 +760,24 @@ public PlanNode visitApply(ApplyNode node, RewriteContext<Set<Symbol>> context)
             PlanNode input = context.rewrite(node.getInput(), inputContext);
             return new ApplyNode(node.getId(), input, subquery, subqueryAssignments.build(), newCorrelation);
         }
+
+        @Override
+        public PlanNode visitLateralJoin(LateralJoinNode node, RewriteContext<Set<Symbol>> context)
+        {
+            PlanNode subquery = context.rewrite(node.getSubquery(), context.get());
```

Comment:
you can remove `LateralJoinNode` completely when type is `INNER` and subquery is scalar and no subquery symbols are referenced. `ApplyNode` was removed for such scalar subqueries.